tools/codegen/model.py

```diff
@@ -360,6 +360,15 @@ def __post_init__(self) -> None:
             assert arg.annotation == ret.annotation, \
                 "Out arguments must have matching return Tensor; furthermore, " \
                 "the ith-argument needs to correspond to the ith return"
+        # Invariant: we expect out arguments to appear as keyword arguments in the schema.
+        # This means that all mutable returns should be aliased to a keyword argument
+        # (except for "self", which we explicitly don't treat as an out argument because of its use in methods)
+        # See Note [is_out_fn]
+        out_and_self = list(self.out_arguments) + [arg for arg in self.arguments if arg.name == "self"]
+        mutable_returns = [ret for ret in self.returns if ret.annotation is not None and ret.annotation.is_write]
+        for ret in mutable_returns:
+            assert any([ret.annotation == arg.annotation for arg in out_and_self]), \
+                "All mutable returns must be aliased either to a keyword argument, or to \"self\""
```

Comment:
This check does something a little stronger than the title suggests- it'll also error out in the following scenario:

```
- func: _compute_linear_combination.out(Tensor input, Tensor coefficients, *, Tensor out) -> Tensor(a!)
```

It's a little more verbose than I'd like. Mostly because `self` gets special treatment (we expect all mutable returns to be aliased to kwargs- except for "self" because of its use in methods. We explicitly DONT want to treat it as an out argument). But I figured that's worth the extra check. E.g. before, only the second of these two checks would fail (even though they seem kind of symmetrically wrong).

```
- func: _compute_linear_combination.out(Tensor input, Tensor coefficients, *, Tensor out) -> Tensor(a!)
- func: _compute_linear_combination.out(Tensor input, Tensor coefficients, *, Tensor out(a!)) -> Tensor
```

Reply:
Might be helpful to add, "Did you forget to mark an out argument is keyword only?"